packages/stateless/src/lib/content/content.engine.ts

```diff
@@ -24,10 +27,21 @@ type HandlebarsContext = {
     | Record<string, unknown>;
 };
 
-export function compileTemplate(content: string, data: HandlebarsContext) {
-  const template = Handlebars.compile<HandlebarsContext>(content);
+export interface IContentEngine {
+  compileTemplate: (content: string, payload: ITriggerPayload) => string;
+  extractMessageVariables: (content: string) => Array<string>;
+}
+
+export class HandlebarsContentEngine implements IContentEngine {
+  compileTemplate(content: string, payload: ITriggerPayload): string {
+    const template = Handlebars.compile<HandlebarsContext>(content);
 
-  return template(data);
+    return template(payload);
+  }
+
+  extractMessageVariables(content: string): string[] {
+    return getHandlebarsVariables(content);
```

Comment:
Either you move the content of the function here or you remove the export of `getHandlebarsVariables` to avoid misuse and contributors refer directly to use this class from now on.